moveit_setup_assistant/src/widgets/robot_poses_widget.h

```diff
@@ -280,6 +280,3 @@ private Q_SLOTS:
 };
 
 }  // namespace moveit_setup_assistant
-
-// Declare std::string as metatype so we can use it in a signal
-Q_DECLARE_METATYPE(std::string)
```

Comment:
Sure, that this is not needed anymore? You will get runtime errors (only), if you try to use `std::string` in a signal-slot connection.

Reply:
There seems to be a few more usages of `SIGNAL(currentIndexChanged(QString))` which was removed in Qt6 it seems. No further warnings
https://doc.qt.io/qt-5/qcombobox-obsolete.html#currentIndexChanged-1

Reply:
```
[INFO] [1715605273.350145827]: Loaded world robot model.
[INFO] [1715605273.352444332]: Setting Param Server with Robot Description
[INFO] [1715605273.357144375]: Robot semantic model successfully loaded.
[INFO] [1715605273.357192642]: Setting Param Server with Robot Semantic Description
qt.core.qobject.connect: QObject::connect: No such signal QButtonGroup::buttonClicked(int)
[INFO] [1715605273.371403570]: Loading robot model 'world'...
[INFO] [1715605273.371460611]: No root/virtual joint specified in SRDF. Assuming fixed joint
qt.core.qobject.connect: QObject::connect: No such signal QComboBox::currentIndexChanged(const QString&)
[INFO] [1715605273.842589137]: Stereo is NOT SUPPORTED
[INFO] [1715605273.842732216]: OpenGL device: RENOIR (renoir, LLVM 15.0.7, DRM 3.54, 6.5.0-28-generic)
[INFO] [1715605273.842779572]: OpenGl version: 4.6 (GLSL 4.6) limited to GLSL 1.4 on Mesa system.
[INFO] [1715605273.977146641]: Loading robot model 'world'...
[INFO] [1715605273.977204534]: No root/virtual joint specified in SRDF. Assuming fixed joint
[INFO] [1715605274.755965943]: Loading Setup Assistant Complete
[INFO] [1715605359.031448956]: Updating kinematic model
[INFO] [1715605359.031652687]: Loading robot model 'world'...
[INFO] [1715605359.031679892]: No root/virtual joint specified in SRDF. Assuming fixed joint
```

I mixed two observations in my last comment:
* There is no runtime warning about std::string signals
* There are broken signal-slot connections, will fix these with another commit

I found evidence on Qt forum that `Q_DECLARE_METATYPE` might only be needed for queued connections or maybe not at all any longer. I turned the connection into a `Qt::QueuedConnection` and it actually still works. Need to test this with Qt5.